applications/FluidDynamicsApplication/python_scripts/navier_stokes_base_solver.py

```diff
@@ -15,132 +15,39 @@ def CreateSolver(main_model_part, custom_settings):
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.MESH_VELOCITY)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.IS_STRUCTURE)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.BODY_FORCE)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_H)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NODAL_AREA)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.REACTION)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.REACTION_WATER_PRESSURE)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.FLAG_VARIABLE)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NORMAL)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.Y_WALL)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DYNAMIC_TAU) # Variable stored in cfd_variables.h
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.OSS_SWITCH)  # Variable stored in cfd_variables.h
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.M)           # Variable stored in cfd_variables.h
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.ADVPROJ)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DIVPROJ)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosCFD.PATCH_INDEX)          # PATCH_INDEX belongs to FluidDynamicsApp.
-        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.EXTERNAL_PRESSURE)
-        self.main_model_part.AddNodalSolutionStepVariable(KratosCFD.Q_VALUE)          # Q_VALUE belongs to FluidDynamicsApp.
-
-
-        print("Base class fluid solver variables added correctly")
+        raise Exception("Trying to add Navier-Stokes base solver variables. Implement the AddVariables() method in the specific derived solver.")
 
     def ImportModelPart(self):
         ## Read model part
-        self._ModelPartReading()
+        self._model_part_reading()
         ## Replace elements and conditions, check the input reading and set KINEMATIC_VISCOSITY and DENSITY
-        self._ExecuteAfterReading()
+        self._execute_after_reading()
         ## Set buffer size
-        self._SetBufferSize()
+        self._set_buffer_size()
 
-        print ("Base class model reading finished.")
+        KratosMultiphysics.Logger.PrintInfo("NavierStokesBaseSolver", "Model reading finished.")
 
     def ExportModelPart(self):
-        name_out_file = self.settings["model_import_settings"]["input_filename"].GetString()+".out"
-        file = open(name_out_file + ".mdpa","w")
-        file.close()
-
         ## Model part writing
+        name_out_file = self.settings["model_import_settings"]["input_filename"].GetString()+".out"
```

Comment:
what is a bit strange to me is that you don't store the settings in the constructor

Reply:
This is rarely used. Having that in mind, I think it's better to avoid creating a self.name_out_file.